packages/core/src/routes/session/continue.test.ts

```diff
@@ -0,0 +1,106 @@
+import dayjs from 'dayjs';
+import { Provider } from 'oidc-provider';
+
+import { mockUser } from '@/__mocks__';
+import { createRequester } from '@/utils/test-utils';
+
+import continueRoutes, { continueRoute } from './continue';
+
+const checkRequiredProfile = jest.fn();
+jest.mock('./utils', () => ({
+  ...jest.requireActual('./utils'),
+  checkRequiredProfile: () => checkRequiredProfile(),
+}));
+
+jest.mock('@/queries/sign-in-experience', () => ({
+  findDefaultSignInExperience: jest.fn(),
+}));
+
+const updateUserById = jest.fn(async (..._args: unknown[]) => mockUser);
+
+jest.mock('@/queries/user', () => ({
+  updateUserById: async (...args: unknown[]) => updateUserById(...args),
+}));
+
+const interactionResult = jest.fn(async () => 'redirectTo');
+const interactionDetails: jest.MockedFunction<() => Promise<unknown>> = jest.fn(async () => ({}));
+
+jest.mock('oidc-provider', () => ({
+  Provider: jest.fn(() => ({
+    interactionDetails,
+    interactionResult,
+  })),
+}));
+
+afterEach(() => {
+  interactionResult.mockClear();
+});
+
+describe('session -> continueRoutes', () => {
+  const sessionRequest = createRequester({
+    anonymousRoutes: continueRoutes,
+    provider: new Provider(''),
+    middlewares: [
+      async (ctx, next) => {
+        ctx.addLogContext = jest.fn();
+        ctx.log = jest.fn();
+
+        return next();
+      },
+    ],
+  });
+
+  describe('POST /session/sign-in/continue/password', () => {
+    it('updates user password, checks required profile, and sign in', async () => {
+      interactionDetails.mockResolvedValueOnce({
+        jti: 'jti',
+        result: {
+          continueSignIn: {
+            userId: mockUser.id,
+            expiresAt: dayjs().add(1, 'day').toISOString(),
+          },
+        },
+      });
+      const response = await sessionRequest.post(`${continueRoute}/password`).send({
+        password: 'password',
+      });
+      expect(checkRequiredProfile).toHaveBeenCalled();
+      expect(response.statusCode).toEqual(200);
+      expect(updateUserById).toHaveBeenCalledWith(mockUser.id, expect.anything());
+      expect(response.body).toHaveProperty('redirectTo');
+      expect(interactionResult).toHaveBeenCalledWith(
+        expect.anything(),
+        expect.anything(),
+        expect.objectContaining({ login: { accountId: mockUser.id } }),
+        expect.anything()
+      );
+    });
```

Comment:
As this mockUser already has a password, not sure this is valid in a real case.

Reply:
If password already exists, then checkProfileRequired should not write a continueSignIn payload into session storage.

Reply:
Yeah... updated.